### PR TITLE
Crafting-gui - colorize book enumeration

### PR DESCRIPTION
**Pull request by Pupsi-Mupsi**
#### Summary

`SUMMARY: Interface "Crafting-gui - colorize book enumeration"`

#### Purpose of change

Make it easier to see which books contain a certain recipe.

Before:
grafik

After:
grafik

### CODE REVIEW

File: src/crafting_gui.cpp [hunk 1]
```diff
@@ -508,7 +508,9 @@ const recipe *select_crafting_recipe( int &batch_size )
                 []( itype_id type_id ) {
                     return item::find_type( type_id )->nname( 1 );
                 } );
-                const std::string text = string_format( _( "Written in: %s" ), enumerated_books.c_str() );
+                const std::string text = string_format( 
+                                            _( "Written in: <color_cyan>%s</color>" ),
```
Comment (AMurkin) [inline src/crafting_gui.cpp]:
```suggestion
                                             _( "Written in: <color_cyan>%s</color>" ), enumerated_books.c_str() );
```

Comment (Pupsi-Mupsi) [reply src/crafting_gui.cpp]: thanks. but still...

File: src/crafting_gui.cpp [hunk 2]
```diff
@@ -508,7 +508,8 @@ const recipe *select_crafting_recipe( int &batch_size )
                 []( itype_id type_id ) {
                     return item::find_type( type_id )->nname( 1 );
                 } );
-                const std::string text = string_format( _( "Written in: %s" ), enumerated_books.c_str() );
+                const std::string text = string_format(
```
Comment (AMurkin) [inline src/crafting_gui.cpp]:
```suggestion
                const std::string text = string_format(
```

Comment (AMurkin) [reply src/crafting_gui.cpp]: Didn't spot this, sorry.

Comment (Pupsi-Mupsi) [reply src/crafting_gui.cpp]: Thank you! You saved me!

**Comment by jbytheway** [comment]
I'm confused as to how this can produce the output pictured.  Based on the code, it looks like it would highlight the whole list in cyan (including the commas, etc.), not just the book names.

**Comment by Pupsi-Mupsi** [comment]
> 
> 
> I'm confused as to how this can produce the output pictured. Based on the code, it looks like it would highlight the whole list in cyan (including the commas, etc.), not just the book names.

What would you change?

**Comment by Pupsi-Mupsi** [comment]
@jbytheway

Will this work.

Sorry I'm not used to write C++.


            if( !g->u.knows_recipe( current[line] ) ) {
                component_print_buffer.push_back( _( "Recipe not memorized yet" ) );
                auto books_with_recipe = g->u.get_books_for_recipe( crafting_inv, current[line] );
                std::string enumerated_books =
                    enumerate_as_string( books_with_recipe.begin(), books_with_recipe.end(),
                []( itype_id type_id ) {
                    return item::find_type( type_id )->nname( 1 );
                } );
                std::string enumerated_colored_books_pre = string_replace( enumerated_books, ", ", "</color>, <color_cyan>" );
                std::string enumerated_colored_books_post = string_replace( enumerated_colored_books_pre, ", and", "</color>, and<color_cyan>" );
                const std::string text = string_format(
                                             _( "Written in: <color_cyan>%s</color>" ), enumerated_colored_books_post.c_str() );
                std::vector<std::string> folded_lines = foldstring( text, pane );
                component_print_buffer.insert(
                    component_print_buffer.end(), folded_lines.begin(), folded_lines.end() );
            }

**Comment by Pupsi-Mupsi** [comment]
> 
> 
> > What would you change?
> 
> The simplest way to do this is to colorize the names before they are concatenated. That can be done like this:
> 
> [jbytheway@b2945e7](https://github.com/jbytheway/Cataclysm-DDA/commit/b2945e72df676de789fd5472d286728b15468753)
> 
> I was going to make this change as a PR onto your branch, but I can't actually seem to make that work for some reason. And also, it's a very small change to the original code (just adding one call to colorize) so to keep things tidy I suggest you close this PR and start afresh with that change instead.

Thank you!!!!

**Comment by ZhilkinSerg** [comment]
> so to keep things tidy I suggest you close this PR and start afresh with that change instead.

I've just squashed commits before merging.

**Pull request merged**